tasks/actorstate/power.go

```diff
@@ -25,46 +27,102 @@ func init() {
 	Register(sa2builtin.StoragePowerActorCodeID, StoragePowerExtractor{})
 }
 
+func NewPowerStateExtractionContext(ctx context.Context, a ActorInfo, node ActorStateAPI) (*PowerStateExtractionContext, error) {
+	curActor, err := node.StateGetActor(ctx, a.Address, a.TipSet)
+	if err != nil {
+		return nil, xerrors.Errorf("loading current power actor: %w", err)
+	}
+
+	curTipset, err := node.ChainGetTipSet(ctx, a.TipSet)
+	if err != nil {
+		return nil, xerrors.Errorf("loading current tipset: %w", err)
+	}
+
+	curState, err := power.Load(node.Store(), curActor)
+	if err != nil {
+		return nil, xerrors.Errorf("loading current power state: %w", err)
+	}
+
+	prevState := curState
+	if a.Epoch != 0 {
+		prevActor, err := node.StateGetActor(ctx, a.Address, a.ParentTipSet)
```

Comment:
has this been fixed in lotus to not be a VM run?

Reply:
:vomiting_face: ughh probably not. This isn't an issue for the API lens (since lotus has a cache) but it will be expensive in the repolens, thanks for catching this. 

Is the canonical way of dealing with this to load the state tree like [this](https://github.com/filecoin-project/sentinel-visor/blob/master/tasks/message/message.go#L331)?

Reply:
for consistency that seems like a good plan

Reply:
I'd like to avoid implementing this logic everywhere we call `StateGetActor` and have filed #196  to track/eventually address this concern. I propose we address this as a fix for #196, thoughts?

Reply:
i'm fine with that